### PR TITLE
Allowing underscores in numbers for easy readability

### PR DESCRIPTION
**Pull request by frroossst**
- just tested with whole numbers not floats
- need to squash commits, we talked about this the last time, but I believe you can configure the repo to allow squashing before merging [link](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/configuring-pull-request-merges/configuring-commit-squashing-for-pull-requests)

### CODE REVIEW

File: util/ascii/ascii.go
Comment (apmckinlay) [inline util/ascii/ascii.go:1]: I'm not sure if this is general enough to go in ascii
I would just make it private (uncapitalized) in lexer.go

File: compile/lexer/lexer.go
Comment (apmckinlay) [inline compile/lexer/lexer.go:1]: To handle underscores before and after the decimal, how about:

lxr.matchWhile(isDigitOrUnderscore)
if lxr.match('.') {
    lxr.matchWhile(IsDigitOrUnderscore)
}

i.e. just change IsDigit to IsDigitOrUnderscore
That allows a trailing underscore which isn't useful but probably ok.

File: util/ascii/ascii.go
Comment (frroossst) [reply util/ascii/ascii.go:1]: that's fair, will make the change

File: compile/lexer/lexer.go
Comment (frroossst) [reply compile/lexer/lexer.go:1]: made the change :+1:

File: builtin/number.go
Comment (apmckinlay) [inline builtin/number.go:1]: I just checked the implementation of ReplaceAll and it already checks if the replacement occurs, so we can eliminate the if's around them.

Comment (frroossst) [reply builtin/number.go:1]: would you like me to remove the ifs around the other places where ReplaceAll occurs too?

**Comment by apmckinlay** [comment]
I would just do the one in your change and the one right next to it.
I can look for any others.

File: builtin/number.go
Comment (frroossst) [reply builtin/number.go:1]: Okay, I've made the change in the latest commit, and from a quick grep search it doesn't look like there are any such instances.

**Comment by apmckinlay** [comment]
I think it looks good to go, should I merge it?

**Comment by frroossst** [comment]
yes please!

**Pull request merged**